core/src/datayoga_core/write_utils.py

```diff
@@ -1,92 +1,52 @@
 import logging
 from itertools import groupby
-from typing import Any, Callable, Dict, List, Union
+from typing import Any, Dict, List, Union
 
-from datayoga_core.block import Block
+from datayoga_core import utils
 from datayoga_core.opcode import OpCode
-from datayoga_core.result import Result, Status
 
 logger = logging.getLogger("dy")
 
 
-def write_records(data: List[Dict[str, Any]],
-                  opcode_field: str, execute_upsert: Callable, execute_delete: Callable) -> List[Dict[str, Any]]:
-    # group records by opcode
-    opcodes = [{"opcode": key, "records": list(result)} for key, result in groupby(
+def group_records_by_opcode(data: List[Dict[str, Any]],
```

Comment:
can we just return 3 return variables instead of a list? is it always just three? In that case, would be easier than a return type of list

Reply:
done